File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var insertIntoBST = function (root, val) {
    let node = root
    let new_node = new TreeNode(val)
    if (!root) return new_node

    while (node != null) {
        if (new_node.val > node.val) {
            if (node.right == null) {
                node.right = new_node
            }
            else {
                node = node.right
            }
        }
        else if (new_node.val < node.val) {
            if (node.left == null) {
                node.left = new_node
            }
            else {
                node = node.left
            }
        }
        else {
            return root
        }
    }
    return root
};

File: 0784-insert-into-a-binary-search-tree/0784-insert-into-a-binary-search-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right

class Solution:
    def insertIntoBST(self, root: Optional[TreeNode], val: int) -> Optional[TreeNode]:
        new_node = TreeNode(val)
        if not root:
            return new_node
        
        node = root
        while node:
            if val > node.val:
                if not node.right:
                    node.right = new_node
                    break
                else:
                    node = node.right
            elif val < node.val:
                if not node.left:
                    node.left = new_node
                    break
                else:
                    node = node.left
            else:
                break
        return root
        
            





